rmw_zenoh_cpp/src/detail/rmw_client_data.cpp

```diff
@@ -40,72 +43,11 @@
 #include "rmw/get_topic_endpoint_info.h"
 #include "rmw/impl/cpp/macros.hpp"
 
-namespace
-{
-
-///=============================================================================
-void client_data_handler(z_loaned_reply_t * reply, void * data)
```

Comment:
Keep these functions for the initial port. We will switch to lambda captures in the follow up PR.

Reply:
Yes, agreed.  This is tricky to get right, so I'd rather do it in a follow-up.